webapp/src/ts/services/create-user-for-contacts.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from '@angular/core';
+import { DbService } from '@mm-services/db.service';
+import { DBSyncService, SyncStatus } from '@mm-services/db-sync.service';
+import { SessionService } from '@mm-services/session.service';
+import { SettingsService } from '@mm-services/settings.service';
+import { UserSettingsService } from '@mm-services/user-settings.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CreateUserForContactsService {
+  constructor(
+    private settingsService: SettingsService,
+    private userSettingsService: UserSettingsService,
+    private dbService: DbService,
+    private dbSyncService: DBSyncService,
+    private sessionService: SessionService,
+  ) {
+    const that = this;
+    this.settingsService
+      .get()
+      .then((settings) => {
+        if (settings?.transitions?.create_user_for_contacts) {
+          that.dbSyncService.subscribe(async(status) => {
+            return that.syncStatusChanged(status);
+          });
+        }
+      });
+  }
+
+  async getUserContact() {
```

Comment:
If we cache it, how would we ever know if the contact becomes replaced?  (Pouch returns document _values_ right? Not just _reverence_ to internal objects that it maintains as state?)  I guess we could maybe just listen for changes to the contact doc... :thinking:

Reply:
Yea, I suppose it's too much effort for something that runs infrequently. 
Though it's a pretty obvious thing to cache, since it's so central to how everything functions in the frontend. 
We even already have a service `UserContactService` that gets this contact. Maybe we could just reuse that one (sans the hydration) and maybe improve that in the future to cache the value.